public/app/features/dashboard/containers/EmbeddedDashboardPage.tsx

```diff
@@ -0,0 +1,148 @@
+  uid: string;
+}
+
+interface EmbeddedDashboardPageRouteSearchParams {
+  callbackUrl?: string;
+  json?: string;
+  accessToken?: string;
+}
+
+export type Props = GrafanaRouteComponentProps<
+  EmbeddedDashboardPageRouteParams,
+  EmbeddedDashboardPageRouteSearchParams
+>;
+
+export default function EmbeddedDashboardPage({ route, queryParams }: Props) {
+  const dispatch = useDispatch();
+  const context = useGrafana();
+  const dashboardState = useSelector((store) => store.dashboard);
+  const dashboard = dashboardState.getModel();
+  const [dashboardJson, setDashboardJson] = useState('');
+
+  /**
+   * Create dashboard model and initialize the dashboard from JSON
+   */
+  useEffect(() => {
+    const callbackUrl = queryParams.callbackUrl;
+
+    if (!callbackUrl) {
+      throw new Error('No callback URL provided');
+    }
+    getBackendSrv()
+      .get(`${callbackUrl}/load-dashboard`)
+      .then((dashboardJson) => {
+        setDashboardJson(dashboardJson);
+        // Remove dashboard UID from JSON to prevent errors from external dashboards
+        delete dashboardJson.uid;
+        const dashboardModel = new DashboardModel(dashboardJson);
+
+        dispatch(
+          initDashboard({
+            routeName: route.routeName,
+            fixUrl: false,
+            keybindingSrv: context.keybindings,
+            dashboardDto: { dashboard: dashboardModel, meta: { canEdit: true } },
+          })
+        );
+      })
+      .catch((err) => {
+        console.log('Error getting dashboard JSON: ', err);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!dashboard) {
+    return <DashboardLoading initPhase={dashboardState.initPhase} />;
+  }
+
+  if (dashboard.meta.dashboardNotFound) {
+    return <p>Not available</p>;
+  }
+
+  return (
+    <Page pageNav={{ text: dashboard.title }} layout={PageLayoutType.Custom}>
+      <Toolbar dashboard={dashboard} callbackUrl={queryParams.callbackUrl} dashboardJson={dashboardJson} />
+      {dashboardState.initError && <DashboardFailed initError={dashboardState.initError} />}
+      <div>
+        <DashboardGrid dashboard={dashboard} isEditable viewPanel={null} editPanel={null} hidePanelMenus />
+      </div>
+    </Page>
+  );
+}
+
+interface ToolbarProps {
+  dashboard: DashboardModel;
+  callbackUrl?: string;
+  dashboardJson: string;
+}
+
+const Toolbar = ({ dashboard, callbackUrl, dashboardJson }: ToolbarProps) => {
+  const dispatch = useDispatch();
+  const styles = useStyles2(getStyles);
+
+  const onChangeTimeZone = (timeZone: TimeZone) => {
+    dispatch(updateTimeZoneForSession(timeZone));
+  };
+
+  const saveDashboard = async (clone: DashboardModel) => {
+    if (!clone || !callbackUrl) {
+      return;
+    }
+
+    return getBackendSrv().post(`${callbackUrl}/save-dashboard`, { dashboard: clone });
+  };
+
+  return (
+    <PageToolbar title={dashboard.title} buttonOverflowAlignment="right" className={styles.toolbar}>
```

Comment:
This is using the deprecated old PageToolbar component from the old nav, not sure we want to keep using it